doc/mgmt/SONiC_Design_Doc_Unified_FRR_Mgmt_Interface.md

```diff
@@ -0,0 +1,1069 @@
+max_prefix_warning_threshold   = 1*3DIGIT; Threshold value (%) at which to generate a warning msg
+max_prefix_restart_interval    = 1*5DIGIT; Restart bgp connection after limit is exceeded
+route_map_in                   = 1*64VCHAR ; Apply route map on incoming routes from neighbor
+route_map_out                  = 1*64VCHAR ; Apply route map on outgoing routes to neighbor
+soft_reconfiguration_in        = "true" / "false" ; Per neighbor soft reconfiguration
+unsuppress_map_name            = 1*64VCHAR ; Route-map to selectively un-suppress suppressed routes        
+route_reflector_client         = "true" / "false" ; Configure a neighbor as Route Reflector client
+weight                         = 1*5DIGIT ; {0..65535} Set default weight for routes from this neighbor
+as_override                    = "true" / "false" ;  Override ASNs in outbound updates if aspath equals remote-as
+send_community                 = "standard" / "extended" / "both" / "none" ; Send Community attribute to this neighbor
+add_path_tx_all                = "true" / "false" ;
+add_path_tx_bestpath           = "true" / "false" ;
+unchanged_as_path              = "true" / "false" ;
+unchanged_med                  = "true" / "false" ;
+unchanged_nexthop              = "true" / "false" ;      
+filter_list_name               = 1*64VCHAR ;
+filter_list_direction          = "inbound" / "outbound";
+nexthop_self_enabled           = "true" / "false" ;
+nexthop_self_force             = "true" / "false" ;       
+prefix_list_name               = 1*64VCHAR ;
+prefix_list_direction          = "inbound" / "outbound";
+remove_private_as_enabled      = "true" / "false" ;
+replace_private_as             = "true" / "false" ;
+remove_private_as_all          = "true" / "false" ;  
+allow_as_count                 = 1*3DIGIT ;  Number of occurences of AS number
+allow_asin                     = "true" / "false" ;  Accept as-path with my AS present in it
+allow_as_origin                = "true" / "false" ; Only accept my AS in the as-path if the route was originated in my AS
+capability_orf_send            = "true" / "false" ; Capability to receive the outbound route filtering from this neighbor
+capability_orf_receive         = "true" / "false" ; Capability to send the outbound route filtering to this neighbor
+capability_orf_both            = "true" / "false" ; Capability to send and receive the outbound route filtering to/from this neighbor
+route-server-client            = "true" / "false" ; Configure a neighbor as Route Server client
+```
+
+#### 3.2.1.6 BGP_PEER_GROUP
+The existing BGP_PEER_RANGE (peer group) table does not have vrf-name as the key (but added as the field in the table as per VRF HLD) but FRR and open config models have the peer-group under VRF context, so, the new table BGP_PEER_GROUP has been introduced for configurations from management framework (This is not a backward compatible change, expecting the user to migrate to this table in the near future).
```

Comment:
BGP_PEER_RANGE is about dynamic bgp peers, it is not about bgp peer groups. SONiC doesn't have peer groups defined in DBs

Reply:
If this table is not for static peer-groups, not sure why we're using this table to generate the peer-group configs in the below file.

https://github.com/Azure/sonic-buildimage/blob/master/dockers/docker-fpm-frr/bgpd.conf.default.j2

{% if BGP_PEER_RANGE %}
{% for bgp_peer in BGP_PEER_RANGE.values() %}
  neighbor {{ bgp_peer['name'] }} peer-group
  neighbor {{ bgp_peer['name'] }} passive
{% if bgp_peer['peer_asn'] is defined %}
  neighbor {{ bgp_peer['name'] }} remote-as {{ bgp_peer['peer_asn'] }}
{% else %}
  neighbor {{ bgp_peer['name'] }} remote-as {{ constants.deployment_id_asn_map[DEVICE_METADATA['localhost']['deployment_id']] }}
{% endif %}
  neighbor {{ bgp_peer['name'] }} ebgp-multihop 255
  neighbor {{ bgp_peer['name'] }} soft-reconfiguration inbound

Reply:
Not only static peers could have peer-group defined for them.
Check following
https://github.com/Azure/SONiC/wiki/Configuration

Reply:
ok, thanks, will remove this sentence.